### PR TITLE
Bumbleweed move + spores

### PR DESCRIPTION
**Pull request by KendallHarter**
Bumbleweed's move is non-infinite and spores now have a non-static supply.

### CODE REVIEW

File: server/game_app/objects.py [hunk 1]
```diff
@@ -21,14 +22,25 @@ def toJson(self):
...
+      plantWorth = 0
+      #determine strength of soakers
+      for plant in self.game.objects.plants:
+        #make sure player owns plant and that plant is a soaker
+        if plant.owner == self.game.playerID and plant.mutation == 3:
+          plantWorth += plant.strength
+
+      sporesYouShouldHave = self.game.maxSpores / 2
+
+      sporesYouGet = math.ceil((sporesYouShouldHave - plantWorth) * self.game.sporeRate)
```
Comment (Tarnasa) [inline server/game_app/objects.py:36]: Is this supposed to be able to remove spores from the player?

File: server/game_app/objects.py [hunk 2]
```diff
@@ -235,16 +247,22 @@ def uproot(self, x, y):
...
     for plant in self.game.objects.plants:
-      if plant.id != self.id:
-        if (plant.x == x) and (plant.y == y):
-          return 'Turn {}: Your plant {} cannot move on top of another plant.'.format(self.game.turnNumber, self.id)
-        #check if movement location is in the range of an owned spawner too
-        if plant.mutation == spawnerNo and plant.owner == self.game.playerID and not inRange:
-          if self.game.dist(x, y, plant.x, plant.y) <= plant.range:
-            if self.game.dist(self.x, self.y, plant.x, plant.y) <= plant.range:
-              inRange = True
+      if not inRange:
```
Comment (Tarnasa) [inline server/game_app/objects.py:257]: This prevents checking for movement onto an existing plant once inRange is true

Comment (KendallHarter) [inline server/game_app/objects.py:257]: I don't think that it's supposed to remove spores.

Comment (Tarnasa) [inline server/game_app/objects.py:257]: So... you'll change it? :P
If the soaker strength is more than half max-spores, then you will start losing spores.  Or am I misunderstanding something?

Comment (KendallHarter) [inline server/game_app/objects.py:257]: Oh I changed
sporesYouShouldHave = self.game.maxSpores
to
sporesYouShouldHave = self.game.maxSpores / 2
for some reason; removing the / 2 should fix this; I guess I should
actually take a look a code before changing it.

**Pull request merged**